zql/zql.peg

```diff
@@ -482,7 +481,7 @@ ExpressionAssignment
     }
 
 FieldAssignment
-  = l:fieldRefDotOnly __ "=" __ r:fieldRefDotOnly {
+  = l:DotExprText __ "=" __ r:DotExpr {
```

Comment:
(This made me realize that put doesn't support dots on the LHS which is strange... unrelated to this PR so no need to address that here).

Reply:
Indeed.  We're getting closer though... we need a universal cut/column builder that can handle this.  None of these builds support reaching into a sub-field as an lvalue to modify an existing field inside a nested column though the column builder can create entirely new nested columns.

Reply:
Issue #1410